File: algorithm_PS/BEAKJOON/DFS/2606.cpp
#include <iostream>
#include <vector>

using namespace std;

int N,M;
vector<vector<int> > virus;
int visited[101] = {};
int cnt = 0;

void dfs(int x){
	visited[x] = true;
	for(int i = 0; i<virus[x].size(); i++){
		int y = virus[x][i];
		if(visited[y] == false){
			dfs(y);
			cnt++;
		}
	}
	return;
}

int main(){
	ios_base :: sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
	cin >> N >> M;
	
	virus.resize(N+1);
	
	for(int i = 0; i<M; i++){
		int x,y;
		cin >> x >> y;
		
		virus[x].push_back(y);
		virus[y].push_back(x);
	}

	dfs(1);
	
	
	cout << cnt;
}

File: algorithm_PS/BEAKJOON/DFS/2606.py
def dfs(v):
    global cnt
    cnt += 1
    visited[v] = True

    for i in adj[v]:
        if not(visited[i]):
            dfs(i)


n = int(input())
m = int(input())

adj = [[] for _ in range(n+1)]
visited = [False] * (n+1)
cnt = 0

for i in range(m):
    x,y = map(int, input().split(' '))

    adj[x].append(y)
    adj[y].append(x)

dfs(1)

print(cnt-1)